KMIP Defaults changes. Support more than 1 KMIP Provider executions

saikumar-vs opened the pull request:
Dependent PR: https://github.com/Percona-QA/percona-qa/pull/415/files

How to test, set configs to:
COMPONENT_KEYRING_FILE=0
COMPONENT_KEYRING_KMIP=1

saikumar-vs committed:
KMIP Defaults changes. Support to execute more than against more than 1 KMIP Provider

saikumar-vs committed:
Updates for lab machine with sudo access and curl instead of wget with retries

saikumar-vs committed:
Updates fix a check

saikumar-vs committed:
Update URI

ParveezBaig commented on pstress/pstress-run.sh:
```diff
@@ -668,8 +682,14 @@ ctrl-c(){
-  echoit "Copying rundir from ${RUNDIR}/${TRIAL} to ${WORKDIR}/${TRIAL}"
-  mv ${RUNDIR}/${TRIAL}/ ${WORKDIR}/ 2>&1 | tee -a /${WORKDIR}/pstress-run.log
+  if [[ ${COMPONENT_KEYRING_KMIP} -eq 1 ]]; then
+    echoit "Copying rundir from ${RUNDIR}/${TRIAL} to ${WORKDIR}/kmip_${kmip_type}/${TRIAL}"
```
Better to try setting WORKDIR  to ${WORKDIR}/"kmip_${kmip_type}" when ${COMPONENT_KEYRING_KMIP} -eq 1 so and avoid these duplicate copy statements here and below.

saikumar-vs replied:
Thanks Parveez for the review, i have addressed your request with other improvements. Pls take a look

saikumar-vs committed:
Refactor KMIP container management and enhance error handling in pstress-run.sh
-  Address review comment to reduce duplicate code
- Renamed `cleanup_existing_container` to `stop_kmip_container` for clarity.
- Improved logging for stopping active KMIP containers.
- Added checks for sudo access before writing configuration files.
- Enhanced port validation to check for TIME_WAIT state.
- Streamlined error handling in various functions to return errors instead of exiting.
- Updated data directory copying logic to use original WORKDIR for templates.
- Cleaned up RUNDIRs after all trials are complete, ensuring proper resource management.

saikumar-vs committed:
Updates: echo statements with echoit . Remove redundant directory creation in savetrial function and ensure WORKDIR and log subdirectory are created as needed.

saikumar-vs committed:
Update hashicorp Script and fix containers stopping at the end of trials

saikumar-vs committed:
Merge branch 'master' into kmip_defaults

vinperothas commented on pstress/pstress-run.sh:
```diff
@@ -2302,6 +2400,74 @@ elif [[ ${PXC} -eq 1 || ${GRP_RPL} -eq 1 ]]; then
+    done
+else
+    COUNT=0
+    for X in $(seq 1 ${TRIALS}); do
```
It looks like there is a redundant for loop in the next if block that causes pstress_test to execute twice as many times as configured TRIALS. Could you please fix the duplicate block?

saikumar-vs replied:
The two trial loops are mutually exclusive if / else branches, not sequential, i.e:
KMIP on → only the KMIP branch runs (TRIALS per KMIP provider)
KMIP off → only the else branch runs (TRIALS once)

vinperothas reviewed (commented):
Encryption with Fortanix is working as expected. However, as you pointed out, mysqld startup takes considerably longer with encryption enabled. Despite increasing the mysqld startup timeout to 180 seconds, a significant number of pstress trials still fail due to startup timeouts.

saikumar-vs commented:
> Encryption with Fortanix is working as expected. However, as you pointed out, mysqld startup takes considerably longer with encryption enabled. Despite increasing the mysqld startup timeout to 180 seconds, a significant number of pstress trials still fail due to startup timeouts.

Thats correct, infact since this leads to failures, i have add the bug report as a comment and for now disabled Fortanix in the configuration.
